perception/autoware_map_based_prediction/src/path_generator.cpp

```diff
@@ -210,37 +228,65 @@ PredictedPath PathGenerator::generateStraightPath(
 
 PredictedPath PathGenerator::generatePolynomialPath(
   const TrackedObject & object, const PosePath & ref_path, const double duration,
-  const double lateral_duration, const double speed_limit) const
+  const double lateral_duration, const double backlash_width, const double speed_limit) const
 {
   std::unique_ptr<ScopedTimeTrack> st_ptr;
   if (time_keeper_) st_ptr = std::make_unique<ScopedTimeTrack>(__func__, *time_keeper_);
 
   // Get current Frenet Point
   const double ref_path_len = autoware::motion_utils::calcArcLength(ref_path);
-  const auto current_point = getFrenetPoint(object, ref_path, duration, speed_limit);
+  const auto current_point = getFrenetPoint(object, ref_path.at(0), duration, speed_limit);
 
-  // Step1. Set Target Frenet Point
+  // Step 1. Set Target Frenet Point
   // Note that we do not set position s,
   // since we don't know the target longitudinal position
   FrenetPoint terminal_point;
-  terminal_point.s_vel = current_point.s_vel;
+  terminal_point.s_vel = std::hypot(current_point.s_vel, current_point.d_vel);
```

Comment:
You mean that the vehicle should keep its speed norm at the finite state, right? If you agree, just resolve this comment.

Reply:
Yes, it is obvious when the object encountered a curved target lane. The vehicle may keep the speed, not longitudinal speed to some lane direction.